docs/deployment/on-chain_deployment_checklist.md

```diff
@@ -0,0 +1,213 @@
+
+    ``` yaml
+    name: <your_skill_name>
+    author: <author>
+    version: <version>
+    type: skill    
+    (...)
+    models:
+      benchmark_tool:
+        args:
+          log_dir: /logs
+        class_name: YourBenchmarkTool
+
+      params:
+        args:
+          consensus:
+            max_participants: 1
+          setup:
+            all_participants:
+            - - '0x0000000000000000000000000000000000000000'
+            safe_contract_address:
+            - '0x0000000000000000000000000000000000000000'
+          tendermint_url: http://localhost:26657
+          tendermint_com_url: http://localhost:8080
+          service_registry_address: null
+          share_tm_config_on_startup: false
+          on_chain_service_id: null
+          (...)
+        class_name: YourSkillParams
+    ```
+
+=== "aea-config.yaml"
+
+    ``` yaml
+    (...)
+    ---
+    public_id: valory/abci:0.1.0
+    type: connection
+    config:
+      target_skill_id: <author>/<your_skill_name>:<version>
+      host: ${ABCI_HOST:str:localhost}
+      port: ${ABCI_PORT:int:26658}
+      use_tendermint: ${ABCI_USE_TENDERMINT:bool:false}
+    
+    ---
+    public_id: valory/ledger:0.19.0
+    type: connection
+    config:
+      ledger_apis:
+        ethereum:
+          address: ${CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_ADDRESS:str:http://localhost:8545}
+          chain_id: ${CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_CHAIN_ID:int:31337}
+          poa_chain: ${CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_POA_CHAIN:bool:false}
+          default_gas_price_strategy: ${CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_DEFAULT_GAS_PRICE_STRATEGY:str:eip1559}
+           
+    ---
+    public_id: valory/p2p_libp2p_client:0.1.0
+    type: connection
+    config:
+      nodes:
+     - uri: ${P2P_URI:str:acn.staging.autonolas.tech:9005}
+       public_key: ${P2P_PUBLIC_KEY:str:02d3a830c9d6ea1ae91936951430dee11f4662f33118b02190693be835359a9d77}
+     - uri: ${P2P_URI:str:acn.staging.autonolas.tech:9006}
+       public_key: ${P2P_PUBLIC_KEY:str:02e741c62d706e1dcf6986bf37fa74b98681bc32669623ac9ee6ff72488d4f59e8}
+    cert_requests:
+    - identifier: acn
+      ledger_id: ethereum
+      message_format: '{public_key}'
+      not_after: '2023-01-01'
+      not_before: '2022-01-01'
+      public_key: ${P2P_PUBLIC_KEY:str:02d3a830c9d6ea1ae91936951430dee11f4662f33118b02190693be835359a9d77}
+      save_path: .certs/acn_cosmos_9005.txt
+
+    ---
+    public_id: <author>/<your_skill_name>:<version>
+    type: skill    
+    models:
+      benchmark_tool:
+        args:
+          log_dir: ${SKILL_<YOUR_SKILL_NAME>_MODELS_BENCHMARK_TOOL_ARGS_LOG_DIR:str:/benchmarks}
+
+      params:
+        args:
+          consensus:
+            max_participants: ${SKILL_<YOUR_SKILL_NAME>_MODELS_PARAMS_ARGS_CONSENSUS_MAX_PARTICIPANTS:int:4}
+          setup:
+            all_participants: ${SKILL_<YOUR_SKILL_NAME>_MODELS_PARAMS_ARGS_SETUP_ALL_PARTICIPANTS:list:[]}      
```

Comment:
These long environment variable names are not required anymore they're generated automatically below the service level. So these environment variable names needs to be defined at the service level, below the service level they're generated automatically so you just need to define the env placeholder with default value and data type like

```suggestion
            all_participants: ${list:[]}      
```

Reply:
Ok. This only applies to aea-config.yaml ?

Reply:
I think @angrybayblade is working on a config validation command. Keep that in mind as once it is available we should add a reference to it here.

Reply:
@dvilelaf I'll open a ticket once I'm done with the features

@jmoreira-valory Yes